jobs/mongodb-server/templates/bin/pre-start.erb

```diff
@@ -219,24 +215,42 @@ else
 
   # Adding current node to rs
   ${MONGO_CMD} --host <%= p("mongodb.replication.replica_set_name") %>/$Master \
-    ${MONGO_CMD_PARAM} --eval 'rs.add("<%= mongo__current_ip %>:<%= mongo_DbTypeport %>")'
+    ${MONGO_CMD_PARAM} --eval 'rs.add("<%= mongo_current_ip %>:<%= mongo_DbTypeport %>")'
   [ $? -ne 0 ] && exit 1
   # need to wait for complete add before restarting
   _timeout=30
   add_ok="false"
   while [ "$add_ok" == "false" ]
   do
-    add_ok=`${MONGO_CMD} --host <%= mongo__current_ip %>:<%= mongo_DbTypeport %> ${MONGO_CMD_PARAM} \
+    add_ok=`${MONGO_CMD} --host <%= mongo_current_ip %>:<%= mongo_DbTypeport %> ${MONGO_CMD_PARAM} \
       --eval 'rs.isMaster().secondary'`
     if [ "$add_ok" != "true" ]
     then
       sleep 2
       _timeout=$((_timeout-2))
-      [ ${_timeout} -le 0 ] && echo "node <%= mongo__current_ip %>:<%= mongo_DbTypeport %> hasn\'t been added. Timeout reached" && exit 1
+      [ ${_timeout} -le 0 ] && echo "node <%= mongo_current_ip %>:<%= mongo_DbTypeport %> hasn\'t been added. Timeout reached" && exit 1
     fi
   done
 
 fi
+}
+
+
+##################################################################################
+#################               Main                   ###########################
+##################################################################################
+sys_update
+
+export MONGO_CMD_PARAM="--quiet"
+
+<%- if  p("mongodb.node_role") == "sa" %>
+# In standalone mode
+start_bs
+create_admin_users
+<%- else -%>
+# In replicaset mode  
+start_replicaset
```

Comment:
It's unfortunately unclear here that users get properly created as above.
We should refactor that somehow later.